verl/experimental/agent_loop/tool_agent_loop.py

```diff
@@ -30,6 +33,49 @@
 logger.setLevel(os.getenv("VERL_LOGGING_LEVEL", "WARN"))
 
 
+class AgentState(Enum):
+    PENDING = "pending"
+    GENERATING = "generating"
+    PROCESSING_TOOLS = "processing_tools"
+    TERMINATED = "terminated"
+    INTERACTING = "interacting"
+
+
+class AgentData:
+    """Encapsulates all state variables for the agent loop."""
+
+    def __init__(
+        self,
+        messages: list[dict[str, Any]],
+        image_data: Any,
+        metrics: dict[str, Any],
+        request_id: str,
+        tools_kwargs: dict[str, Any],
+        interaction: Optional[BaseInteraction] = None,
+        interaction_kwargs: Optional[dict[str, Any]] = None,
+    ):
+        self.messages = messages
+        self.image_data = image_data
+        self.metrics = metrics
+        self.request_id = request_id
+        self.tools_kwargs = tools_kwargs
+        self.interaction = interaction
+        self.interaction_kwargs = interaction_kwargs or {}
+
+        # State variables
+        self.prompt_ids: list[int] = []
+        self.response_mask: list[int] = []
+        self.response_logprobs: list[float] = []
+        self.turn_scores: list[float] = []
+        self.user_turns = 0
+        self.assistant_turns = 0
+
+        # Temporary state for tool calls
+        self.tool_calls: list[FunctionCall] = []
+        self.response_ids: list[int] = []
```

Comment:
`self.response_ids` is not a `Temporary state for tool calls`, move it under self.prompt_ids.